src/Tools/dotnet-monitor/InProcessFeatures/InProcessFeaturesService.cs

```diff
@@ -30,7 +31,7 @@ public async Task ApplyInProcessFeatures(IEndpointInfo endpointInfo, Cancellatio
             try
             {
                 DiagnosticsClient client = new DiagnosticsClient(endpointInfo.Endpoint);
-                if (_parameterCapturingOptions.GetEnabled())
+                if (_parameterCapturingOptions.GetEnabled() && CaptureParametersOperation.IsEndpointRuntimeSupported(endpointInfo))
```

Comment:
Which version of the profiler is going to get loaded for these circumstances? We should not be loading the mutating profiler if we cannot use parameter capturing.

Reply:
I think we should add what @wiktork was getting at: don't bother to load the mutating profiler if parameter capturing is not applicable; this will help avoid customer reports that may blame us when we really aren't involved in the actual problem. Ideally, we would have a capability and requirement description system to declare this, but we don't at this time, so just add it as an explicit check in the profiler service.